src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntryValidator.cs

```diff
@@ -0,0 +1,170 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Query;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class CustomTagEntryValidator : ICustomTagEntryValidator
+    {
+        private const string UnknownTagName = "Unknown";
+
+        /* Unsupported VRCodes:
+         LT(Long Text), OB (Other Byte), OD (Other Double), OF(Other Float), OL (Other Long), OV(other Very long), OW (other Word), ST(Short Text, SV (Signed Very long)
+         UC (Unlimited Characters), UN (Unknown), UR (URI), UT (Unlimited Text), UV (Unsigned Very long)
+         Note: we dont' find definition for UR, UV and SV in DICOM standard (http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_6.2.html)
+        */
+
+        public static readonly IImmutableSet<string> SupportedVRCodes = ImmutableHashSet.Create(
+            DicomVRCode.AE,
+            DicomVRCode.AS,
+            DicomVRCode.AT,
+            DicomVRCode.CS,
+            DicomVRCode.DA,
+            DicomVRCode.DS,
+            DicomVRCode.DT,
+            DicomVRCode.FD,
+            DicomVRCode.FL,
+            DicomVRCode.IS,
+            DicomVRCode.LO,
+            DicomVRCode.PN,
+            DicomVRCode.SH,
+            DicomVRCode.SL,
+            DicomVRCode.SS,
+            DicomVRCode.TM,
+            DicomVRCode.UI,
+            DicomVRCode.UL,
+            DicomVRCode.US);
+
+        public void ValidateCustomTags(IEnumerable<CustomTagEntry> customTagEntries)
```

Comment:
What is a customTagEntry record? What is stored in our DB or what is passed in from the customer. Because it has a Key and status property and I am not sure how can user pass that it.

We shouls separate user input model from stored entry